### PR TITLE
[AJ-1705] Specify cloning instructions and purpose when linking snapshots through Rawls

### PR DESCRIPTION
**Pull request by nawatts**
https://broadworkbench.atlassian.net/browse/AJ-1705

Part 1 of [AJ-1705](https://broadworkbench.atlassian.net/browse/AJ-1705)...

When we link snapshots via WSM, we specify "REFERENCE" for cloning instructions and "policy" for purpose.
https://github.com/DataBiosphere/terra-workspace-data-service/blob/be53253095d079e52b12b7631663404485e8c861/service/src/main/java/org/databiosphere/workspacedataservice/workspacemanager/WorkspaceManagerDao.java#L41-L77

This passes those same arguments when linking snapshots via Rawls.

Support for these options was added to Rawls in https://github.com/broadinstitute/rawls/pull/2790.

[AJ-1705]: https://broadworkbench.atlassian.net/browse/AJ-1705?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: service/src/test/java/org/databiosphere/workspacedataservice/rawls/RawlsClientTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.mockito.stubbing.Answer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestPropertySource;
+
+@DirtiesContext
+@ActiveProfiles(value = "control-plane", inheritProfiles = false)
```
Comment (calypsomatic) [inline service/src/test/java/org/databiosphere/workspacedataservice/rawls/RawlsClientTest.java:27]: I'm not positive, but I think `inheritProfiles = false` is only needed if you're extending `TestBase`

Comment (nawatts) [reply service/src/test/java/org/databiosphere/workspacedataservice/rawls/RawlsClientTest.java:27]: You're right.

Comment (nawatts) [reply service/src/test/java/org/databiosphere/workspacedataservice/rawls/RawlsClientTest.java]: Removed in https://github.com/DataBiosphere/terra-workspace-data-service/pull/688/commits/28586b90eb094d1085362632b6e605b3664f462a.

File: service/src/test/java/org/databiosphere/workspacedataservice/rawls/RawlsClientTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    UUID snapshotId = UUID.randomUUID();
+
+    when(mockRawlsApi.createDataRepoSnapshotByWorkspaceId(eq(workspaceId.id()), any()))
+        .thenAnswer((Answer<?>) invocation -> null);
+
+    // ACT
+    rawlsClient.createSnapshotReference(workspaceId.id(), snapshotId);
+
+    // ASSERT
+    // verify it retried three times, until it got the success
```
Comment (calypsomatic) [inline service/src/test/java/org/databiosphere/workspacedataservice/rawls/RawlsClientTest.java:61]: Is this comment correct or copy-pasta?

Comment (nawatts) [reply service/src/test/java/org/databiosphere/workspacedataservice/rawls/RawlsClientTest.java:61]: The latter... 😞

**Review by calypsomatic — APPROVED** [review]

**Review by davidangb — APPROVED** [review]

**Pull request merged**